PluginsAndFeatures/azure-toolkit-for-intellij/src/main/java/com/microsoft/azure/toolkit/intellij/appservice/action/SSHIntoWebAppAction.java

```diff
@@ -54,31 +56,49 @@ public SSHIntoWebAppAction(WebAppNode webAppNode) {
     }
 
     @Override
+
     protected void actionPerformed(NodeActionEvent nodeActionEvent) throws AzureCmdException {
         logger.info(message("webapp.ssh.hint.startSSH", webAppName));
         // ssh to connect to remote web app container.
         final AzureString title = title("webapp|ssh.connect", webAppName);
-        AzureTaskManager.getInstance().runInBackground(new AzureTask(project, title, false, () -> {
-            // check these conditions to ssh into web app
-            if (!SSHTerminalManager.INSTANCE.beforeExecuteAzCreateRemoteConnection(subscriptionId, webApp.getRuntime())) {
-                return;
-            }
-            // build proxy between remote and local
-            SSHTerminalManager.CreateRemoteConnectionOutput connectionInfo = SSHTerminalManager.INSTANCE.executeAzCreateRemoteConnectionAndGetOutput(
-                AzureCliUtils.formatCreateWebAppRemoteConnectionParameters(subscriptionId, resourceGroupName, webAppName));
-            logger.info(message("webapp.ssh.hint.sshConnectionDone", connectionInfo.getOutputMessage()));
-            // ssh to local proxy and open terminal.
-            AzureTaskManager.getInstance().runAndWait(() -> {
-                // create a new terminal tab.
-                TerminalView terminalView = TerminalView.getInstance(project);
-                ShellTerminalWidget shellTerminalWidget = terminalView.createLocalShellWidget(null, String.format(WEBAPP_TERMINAL_TABLE_NAME, webAppName));
-                final AzureString messageTitle = title("webapp|ssh.open", webAppName);
-                AzureTaskManager.getInstance().runInBackground(new AzureTask(project, messageTitle, false, () -> {
-                    // create connection to the local proxy.
-                    SSHTerminalManager.INSTANCE.openConnectionInTerminal(shellTerminalWidget, connectionInfo);
-                }));
-            }, AzureTask.Modality.ANY);
-        }));
+        AzureTaskManager.getInstance().runInBackground(new AzureTask(project, title, false,
+            () -> {
+                // check these conditions to ssh into web app
+                if (!SSHTerminalManager.INSTANCE.beforeExecuteAzCreateRemoteConnection(webApp.getRuntime())) {
```

Comment:
as current logic doesn't depend on Azure CLI, the `beforeExecuteAzCreateRemoteConnection` should be rename.

Reply:
it is very simple now. I suggest to delete it and movo the check logic here directly

Reply:
I have already disabled the menu for non-linux webapp